lib/python/qmk/keymap.py

```diff
@@ -20,8 +20,34 @@
 };
 """
 
+DEFAULT_KEYMAP_ENCODERS_C = """#include QMK_KEYBOARD_H
 
-def template(keyboard):
+/* THIS FILE WAS GENERATED AND IS EXPERIMENTAL!
+ *
+ * This file was generated by qmk-compile-json. You may or may not want to
+ * edit it directly.
+ */
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+__KEYMAP_GOES_HERE__
+};
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+__ENCODERS_GO_HERE__
+};
+"""
+
+ENCODER_IF = """\t_ELIF_ (index == _INDEX_) {
+\t\tif (clockwise) {
+\t\t\ttap_code(_CLOCKWISE_);
+\t\t} else {
+\t\t\ttap_code(_COUNTER_);
+\t\t}
+\t}
+"""
```

Comment:
You should append this rather than create a new template. Keyboard maintainers can supply their own templates and this would break that feature.

Reply:
done